### ISSUE TITLE
`Geometry.Point` Equals returns true when Lat/Long on both points are null.

### ISSUE THREAD

**Issue by morehavoc** [dymaptic/GeoBlazor#149]
## TL;DR
I think we need to do two things:
1) Remove the check for duplicate geometries, and just add whatever the dev gives us
2) Figure out the lat/long issue here and fix Equas to work correctly

## And now the _rest_ of the story
I discovered this while adding points to graphics layers. It turns out that GeoBlazor checks to see if the graphic has already been added... I don't think we should do this check. It seems strange to add the same graphic multiple times.. but I can see it as a workflow. Maybe I need to throw them on the map and then be able to do a count when the user clicks? So.. I think we should remove that, or if we need it, only check the ID. It's also a little expensive if I have lots of graphics!

But back to the problem at hand. When comparing two graphics, it checks the `Geometry` and these two points are coming back as `true` (Equal).

Point 1:
![image](https://user-images.githubusercontent.com/1805118/225219842-1a246720-db0e-4456-a3fa-5d11050866f4.png)

Point 2:
![image](https://user-images.githubusercontent.com/1805118/225219880-47f5a210-5ee0-4abd-85f4-9eae91f21916.png)

These two points are _not_ the same. However, they do both have the same Lat/Long values... null.

If I understand this function correctly:

```
public bool Equals(Point? other)
    {
        if (ReferenceEquals(null, other)) return false;
        if (ReferenceEquals(this, other)) return true;

        return (Latitude.Equals(other.Latitude) && Longitude.Equals(other.Longitude)) ||
            (X.Equals(other.X) && Y.Equals(other.Y) && Z.Equals(other.Z));
    }
```

Then `Latitude.Equals(other.Latitude)` when both Latitude's are null will be true. This proves to be true in an immediate window as well:
![image](https://user-images.githubusercontent.com/1805118/225220303-4c680e30-8162-4772-985c-8d1c96253e67.png)

The [docs](https://developers.arcgis.com/javascript/latest/api-reference/esri-geometry-Point.html#latitude) for Lat and Long properties in Javascript say:
> The latitude of the point. If the spatial reference is Web Mercator, the latitude will be given in WGS84. In any geographic spatial reference, the latitude will equal the [y](https://developers.arcgis.com/javascript/latest/api-reference/esri-geometry-Point.html#y) coordinate. In all other cases the latitude will be null.

In my case, I am using geometries that I have done things to, projected, densified, etc. and the Lat/long values are not populated. So, we need to sort out when lat/long should participate in the equality test, if ever.

**Comment by TimPurdum** [comment]
@morehavoc once again the rub comes down to the razor component rendering support. Without an equality check, it is possible for the rendering cycle to continue adding the same graphic to the layer or view, since we are manually registering everything. It is _possible_ that this is no longer an issue given other changes, but that would be my main concern to test, that removing equality checks does not create many duplicates in markup-rendered graphics.

As far as just updating the equality check, I'm 100% for this. Would it work to just do `&&` between Lat/Long and X/Y/Z? If either set is null, it would still check the other set.

**Comment by TimPurdum** [comment]
@morehavoc I'm revisiting this open gissue. It doesn't seem to be impacting users much, however, I would like to resolve it. Based on the doc link you shared above, I think we could do equality checks _only_ on X/Y, and skip Lat/Long, since they are either `null`, or the same as X/Y. Does that make sense?

**Comment by morehavoc** [comment]
@TimPurdum For point geometries, I think we might have to check all of them: Lat, Long, X, Y, Z, M. I think, based on your descriptions of how we have to keep track of things between renders, that we could have points that are Lat, Long, but no other properties, so I think we have to check all 6 values to ensure they are the same (they can be null, as long as they are null in both points).

For other geometries, we should check each point in the list to verify that each one is the same, and that each point needs to be compared on all of its properties (which I think are X, Y, Z, M at most).

**Comment by TimPurdum** [comment]
@morehavoc I've been doing more digging, and you were correct. I fixed the re-rendering issue that is at the root of GeoBlazor awhile ago, and I am in the process of testing removing all the equality checks.
